File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/SalesOrderItemDto.cs
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectPos.Services.DTOs
{
    public class SalesOrderItemDto : EntityDto<int>
    {
        public int? OrderNumber { get; set; }
        public string? ProductName { get; set; }
        public string? BarCode { get; set; }
        public bool? IsTaxable { get; set; }
        public int? ProductId { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Quantity { get; set; }
        public Unit? Unit { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? UnitPrice { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Price { get; set; }
    }
}

File: ProjectPos.FrontOffice.FrontEnd/src/proxy/interfaces/sales-order-item-dto.ts
import { EntityDto } from "../entity-dtos/entity-dto";
import { Category } from "../enums/category";
import { Unit } from "../enums/unit";
import { ProductDto } from "./product-dto";
import { ProductInventoryDto } from "./product-inventory-dto";

export interface SalesOrderItemDto extends EntityDto<number> {
    rowIndex?: number;
    orderNumber?: number;
    productName?: string;
    category?: Category;
    subCategory?: string;
    isTaxable?: boolean;
    barCode?: string;
    productId?: number;
    product?: ProductInventoryDto;
    vat?: number;
    vatPercentage?: number;
    quantity?: number;
    unit?: Unit;
    unitPrice?: number;
    price?: number;
    isReturned?: boolean;
    currencyUnitPrice?: number;
    currencyPrice?: number;
}
